### PR TITLE
[Feature] Organize directories by partition in shared_data cluster

### PR DESCRIPTION
**Pull request by sduzh**
Organize remote storage directories by partition, i.e, each partition has its own subdirectory containing metadata and data files.

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 44 / 49 (89.80%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[161]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|5|60.00%|[163, 167]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|7|85.71%|[502]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|27|27|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 43 / 50 (86.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[161]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|5|60.00%|[163, 167]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|7|85.71%|[502]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|26|28|92.86%|[100, 101]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java [hunk 1]
```diff
@@ -2829,12 +2829,15 @@ public boolean supportInsert() {
     }
 
     // ------ for lake table and lake materialized view start ------
-    public String getStoragePath() {
-        throw new SemanticException("getStoragePath is not supported");
+    public FilePathInfo getDefaultFilePathInfo() {
+        return tableProperty.getStorageInfo().getFilePathInfo();
```
[the remark this answers is not in the document]
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java]: Updated

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 47 / 54 (87.04%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[161]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|5|60.00%|[163, 167]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|7|85.71%|[502]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|26|28|92.86%|[100, 101]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 51 / 58 (87.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[161]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|5|60.00%|[163, 167]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|7|85.71%|[502]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|30|32|93.75%|[104, 105]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 51 / 58 (87.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[161]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|5|60.00%|[163, 167]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|7|85.71%|[502]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|30|32|93.75%|[104, 105]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|1|1|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 51 / 58 (87.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[161]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|5|60.00%|[163, 167]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|7|85.71%|[502]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|30|32|93.75%|[104, 105]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 51 / 58 (87.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[161]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|5|60.00%|[163, 167]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|7|85.71%|[502]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|30|32|93.75%|[104, 105]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 51 / 58 (87.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[161]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|5|60.00%|[163, 167]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|7|85.71%|[502]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|30|32|93.75%|[104, 105]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 51 / 58 (87.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[227]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[161]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|5|60.00%|[163, 167]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|7|85.71%|[502]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|30|32|93.75%|[104, 105]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|1|1|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**